doc/whatsnew/2/2.14/summary.rst

```diff
@@ -260,6 +260,8 @@ Other Changes
 
   Closes #6467
 
+* Fix type annotations of class and instance attributes using the alternative union syntax.
+
```

Comment:
@Pierre-Sassoulas from what I've read in the new PR template this should be the right place for a user-facing bug fix.

Reply:
If it's in summary.py it should also be in the full changelog for 2.14 (we're going to release soon so it's not going to last). For 2.15 we're discussing what we're going to do in #6688, right now it's "all in a single file, in the right section (to be determined), with the PR number (not the original issue number) and enforced by the CI if the PR is not labelled maintenance or documentation in github". If you have an idea on how to do that better that would be welcome 😄